src/solvers/lowering/expr_lowering.h

```diff
@@ -11,9 +11,19 @@ Author: Michael Tautschnig
 
 #include <util/expr.h>
 
+class byte_extract_exprt;
+class byte_update_exprt;
 class namespacet;
 class popcount_exprt;
 
+exprt flatten_byte_extract(const byte_extract_exprt &src, const namespacet &ns);
```

Comment:
If `flattening` is different from `lowering` then why are the functions in `lowering` called `flatten`ing?

Reply:
They will be renamed (I had already done so in https://github.com/diffblue/cbmc/pull/2068/commits/8cd566ef6d3d378c55629c67c4594cb04ae1e183), but I wanted to keep this one as easy as possible to review. I'll follow up with a PR doing the renaming shortly.